src/main/webapp/app/view/project/admin/locations/project-locations.component.ts

```diff
@@ -1,8 +1,122 @@
+import { combineLatest, Subscription } from 'rxjs';
+import { JhiEventManager } from 'ng-jhipster';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { Project } from 'app/shared/model/project.model';
+import { Location } from 'app/shared/model/location.model';
+
+import { LocationService } from 'app/entities/location/location.service';
+import { ProjectLocationDeleteDialogComponent } from 'app/view/project/admin/locations/project-location-delete-dialog.component';
+
+import { ITEMS_PER_PAGE } from 'app/shared/constants/pagination.constants';
 
 @Component({
   selector: 'app-project-locations',
   templateUrl: './project-locations.component.html',
   styleUrls: ['project-locations.component.scss'],
 })
-export class ProjectLocationsComponent {}
+export class ProjectLocationsComponent implements OnInit, OnDestroy {
+  project?: Project;
+  locations?: Location[];
+
+  eventSubscriber?: Subscription;
+
+  totalItems = 0;
+  itemsPerPage = ITEMS_PER_PAGE;
+  page!: number;
+  predicate = 'name';
+  ascending = true;
+  ngbPaginationPage = 1;
+
+  constructor(
+    protected locationService: LocationService,
+    protected activatedRoute: ActivatedRoute,
+    protected router: Router,
+    protected eventManager: JhiEventManager,
+    protected modalService: NgbModal
+  ) {}
+
+  ngOnInit(): void {
+    this.activatedRoute.data.subscribe(({ project }) => {
+      this.project = project;
+      this.handleNavigation();
+      this.registerChangeInLocations();
+    });
+  }
+
+  ngOnDestroy(): void {
+    if (this.eventSubscriber) {
+      this.eventManager.destroy(this.eventSubscriber);
+    }
+  }
+
+  protected handleNavigation(): void {
+    combineLatest(this.activatedRoute.data, this.activatedRoute.queryParamMap, (data: Data, params: ParamMap) => {
+      const page = params.get('page');
+      const pageNumber = page !== null ? +page : 1;
+      if (pageNumber !== this.page) {
+        this.loadPage(pageNumber, true);
+      }
+    }).subscribe();
+  }
+
+  loadPage(page?: number, dontNavigate?: boolean): void {
+    const pageToLoad: number = page || this.page || 1;
+
+    this.locationService
+      .findAllByProject(this.project!, {
+        page: pageToLoad - 1,
+        size: this.itemsPerPage,
+        sort: this.sort(),
+      })
+      .subscribe(
+        (res: HttpResponse<Location[]>) => this.onSuccess(res.body, res.headers, pageToLoad, !dontNavigate),
+        () => this.onError()
+      );
+  }
+
+  trackId(index: number, item: Location): number {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    return item.id!;
+  }
+
+  registerChangeInLocations(): void {
+    this.eventSubscriber = this.eventManager.subscribe('locationListModification', () => this.loadPage());
+  }
+
+  delete(location: Location): void {
+    const modalRef = this.modalService.open(ProjectLocationDeleteDialogComponent, { size: 'lg', backdrop: 'static' });
+    modalRef.componentInstance.location = location;
```

Comment:
why don't you resolve it via route/id param?

Reply:
Habe ich von project-responsibilities.component.ts übernommen. Wie würde man es mit dem Resolver lösen?

Reply:
naja, dann müsste die route die id enthalten.. denke, das geht aber auch so.